src/Microsoft.CodeAnalysis.Analyzers/Core/FixAnalyzers/FixerWithFixAllAnalyzer.Fixer.cs

```diff
@@ -53,7 +53,7 @@ private static async Task<Document> AddMethodAsync(Document document, SyntaxNode
             var generator = editor.Generator;
 
             var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var typeIsSealed = ((INamedTypeSymbol)model.GetDeclaredSymbol(classDecl)).IsSealed;
+            var typeIsSealed = ((INamedTypeSymbol)model.GetDeclaredSymbol(classDecl, cancellationToken)).IsSealed;
```

Comment:
This isn't related to this change but is it safe to use a direct cast here? Can't we have a `ITypeSymbol` which won't be a `INamedTypeSymbol`?

Reply:
The analyzer associated with this fixer report diagnostics only for `INamedTypeSymbol`, so I expect the cast is valid:

Here is a snippet from the analyzer:

```
                foreach (INamedTypeSymbol fixer in _codeFixProviders)
                {
                    if (OverridesGetFixAllProvider(fixer))
                    {
                        AnalyzeFixerWithFixAll(fixer, context);
                    }
                    else if (fixer.BaseType != null && fixer.BaseType.Equals(_codeFixProviderSymbol))
                    {
                        Diagnostic diagnostic = Diagnostic.Create(OverrideGetFixAllProviderRule, fixer.Locations.First(), fixer.Name);
                        context.ReportDiagnostic(diagnostic);
                    }
                }
```

Reply:
I would still do safe-cast as the code might have changed between the diagnostic and the codefix registration being called but I might be wrong.